### PR TITLE
More routes and organisation

### PR DESCRIPTION
**Pull request by g1st**
### Proposal
I changed `jobs` table slightly not to include start, end and last codes. Instead added `status` field for simply keeping state 0 as default which means job is not submitted and we will change it to 1 when the job will be submitted by cleaner.

Also added more routes which we built with @GTabala yesterday and extracted them into separate files for sanity.

Backend validation with `express-validator` also added but might be needed to tweak here and there still.

`db.js` file's export changed to the code from [official docs async/await example](https://node-postgres.com/guides/project-structure). As we had before not released connection and our database hanged, so now it will show some messages in console log if there will be something wrong with connections.

### Checklist

- [x] I have made this pull request to the `master` branch
- [x] I have run `npm run lint` and there are no errors

### CODE REVIEW

File: server/routes/jobs.js [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+const router = new Router();
+
+router.get("/jobs", (_, res, next) => {
+	db.query(
+		`SELECT j.id, j.status, j.date_created,b.address, j.visit_on, j.visit_time, j.pay_rate, j.details, j.start_time, j.end_time, c.name customer, w.name worker
+		FROM jobs j 
+		INNER JOIN branches b ON j.branch_id=b.id 
+		INNER JOIN customers c ON j.customer_id=c.id
+		INNER JOIN workers w ON w.id=j.worker_id`
+	)
```
Comment (GTabala) [inline server/routes/jobs.js:14]: I see a new attribute "status". I can't see where you set it up. We can still use checking of certain fields if they are not empty for the calculating status of the document.

File: server/routes/jobs.js [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+		.then(({ rows }) => {
+			return res.json({ jobs: rows });
```
Comment (GTabala) [inline server/routes/jobs.js:16]: I see a new attribute "status". I can't see where you set it up. We can still use checking of certain fields if they are not empty for the calculating status of the document.

It is necessary to provide for the case when the document is saved, but some values have not yet been selected: branch or worker.

File: server/routes/jobs.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+			FROM customers c
+			INNER JOIN branches b ON c.id=b.customer_id
+			INNER JOIN workers w ON w.id=b.worker_id
+			WHERE c.id=$1 AND b.id=c.main_branch_id`,
+			[id]
+		);
+		if (fullJobDetails.rows < 1) {
+			// customer has main branch but no default worker
+			const jobDetailsBranch = await client.query(
+				`
```
Comment (GTabala) [inline server/routes/jobs.js:41]: The same route essential for branches and workers.

File: server/routes/jobs.js (hunk 2 above)
Comment (g1st) [reply server/routes/jobs.js:16]: As I have mentioned in initial PR message, I removed start/end and last code fields from `jobs` table. It makes that table easier to read. I introduced status field for keeping job status. By default it is always set to 0. It changes to 1 only when worker logs his time. Then it becomes 1. Then we dont need to count that status value from other fields if they are empty or not. And we have a dedicated field for that. For me that looks like a good idea but if I am wrong and missing something here I can get rid of that `status` variable and we can compute status if that IF ELSE sql query check.

File: server/routes/branches.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
+import { Router } from "express";
+import { body, validationResult } from "express-validator";
+import db from "../db";
+
+const router = new Router();
+
+router.get("/branches", (_, res, next) => {
```
Comment (Harriethw) [inline server/routes/branches.js:7]: What are `branches` here? Could you update me on the context for this?

File: server/routes/jobs.js (hunk 2 above)
Comment (GTabala) [reply server/routes/jobs.js:16]: I found the place where you set up status. Thank you.

File: server/routes/branches.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+		}
+	} catch (e) {
+		next(e);
+	} finally {
+		client.release();
+	}
+});
+
+router.post(
+	"/branches/:customerId",
```
Comment (Harriethw) [inline server/routes/branches.js:51]: I don't think you'd normally `POST` to a specific ID. A POST request to `/branches` should contain all the data needed to create the `branch`

File: server/spring_action_cleaning.sql [hunk 6]
```diff
@@ -47,11 +47,9 @@ CREATE TABLE jobs (
   details         VARCHAR(1000),
   visit_on        DATE NOT NULL,
   visit_time      TIME (0) NOT NULL,
-  start_code      VARCHAR(4),
+  status          INT NOT NULL DEFAULT 0,
```
Comment (Harriethw) [inline server/spring_action_cleaning.sql:50]: seems like a good simplification for now 👍

File: server/routes/branches.js (hunk 4 above)
Comment (g1st) [reply server/routes/branches.js:7]: Hi @Harriethw sure, branches are addresses. Client can have multiple branches and a single default branch.

File: server/routes/branches.js (hunk 5 above)
Comment (g1st) [reply server/routes/branches.js:51]: We creating a branch (address) for a specific customer. I don't think that we can have branches without a customer. @GTabala can you confirm this?
Also I thought to have this customerID in req.body object, but Gennady said it is more clear to have it as a parameter. For me personaly that doesn't matter I can use both approaches.

**Pull request merged**